### DISCUSSION TITLE
Extracting binary weights and activation values generated after training?

### DISCUSSION THREAD

**Discussion by will-arden** [larq/larq#876] [Q&A]
I am interested in using the BNN tutorial using the MNIST dataset, and was wondering how I am able to find all of the generated values after the training process? My end goal is to write some assembly code to submit some image data (stored somewhere in memory) of a handwritten digit and get the trained model to classify it - I am unsure how to get here as I am very new to neural networks and Larq especially. Any help would be really appreciated!

Edit: typo

**Comment by CNugteren** [comment]
Thank you for your interest in Larq.

Regarding printing out the weights of the trained model, I believe the answer I gave [here](https://github.com/larq/larq/discussions/772#discussioncomment-2805457) will help you out.

But are you sure you want to write your own inference engine (in assembly?). Note that for Cortex-A class devices we provide [LCE, the Larq Compute Engine](https://docs.larq.dev/compute-engine/) which can run optimized BNN kernels. At the very least you can take some inspiration from there.

**Reply by will-arden** [reply]
Hi there, thanks for the reply! Yes I am working on a project involving FPGAs, and since I am so new to neural networks (and I would like to keep the assembly as straightforward as possible) I was wondering if I could create with Larq an extremely simple Binarized NN trained on MNIST (or similar)? Then I could extract the weights and write some assembly for the simple BNN.

Or might it instead be easier to make the simple NN in Keras and post-quantise in Larq?
Many thanks in advance for any advice!

**Reply by CNugteren** [reply]
>  was wondering if I could create with Larq an extremely simple Binarized NN trained on MNIST (or similar)? Then I could extract the weights and write some assembly for the simple BNN.

Yes I would suggest to take that path.
